app/src/main/java/com/skyyo/samples/features/textSpans/composables/UnderlineSpansText.kt

```diff
@@ -0,0 +1,238 @@
+                    val path = Path().apply { buildSquigglesFor(bound, density) }
+                    drawPath(
+                        color = Color.Green,
+                        path = path,
+                        style = pathStyle
+                    )
+                }
+            }
+        }
+    )
+}
+
+data class SimplePaddingValues(val horizontal: Dp, val vertical: Dp)
+
+@Composable
+fun HighlightedText(
+    modifier: Modifier = Modifier,
+    text: AnnotatedString,
+    fontSize: TextUnit = TextUnit.Unspecified,
+    lineHeight: TextUnit = TextUnit.Unspecified,
+    cornerRadius: CornerRadius,
+    highlightColor: Color,
+    highlightBorderColor: Color? = null,
+    highlightBorderWidth: Dp = 2.dp,
+    padding: SimplePaddingValues = remember {
+        SimplePaddingValues(
+            horizontal = 0.dp,
+            vertical = 0.dp
+        )
+    },
+) {
+    val onDraw: MutableState<DrawScope.() -> Unit> = remember { mutableStateOf({}) }
+    Text(
+        modifier = modifier.drawBehind { onDraw.value(this) },
+        text = text,
+        fontSize = fontSize,
+        lineHeight = lineHeight,
+        onTextLayout = { layoutResult ->
+            val annotation = text.getStringAnnotations("squiggles", 0, text.length).first()
+            val textBounds = layoutResult.getBoundingBoxes(annotation.start, annotation.end)
+            onDraw.value = {
+                textBounds.fastForEachIndexed { index, bound ->
+                    val path = Path().apply {
+                        addRoundRect(
+                            RoundRect(
+                                rect = bound.copy(
+                                    left = bound.left - padding.horizontal.toPx(),
+                                    right = bound.right + padding.horizontal.toPx(),
+                                    top = bound.top - padding.vertical.toPx(),
+                                    bottom = bound.bottom + padding.vertical.toPx()
+                                ),
+                                topLeft = if (index == 0) cornerRadius else CornerRadius.Zero,
+                                bottomLeft = if (index == 0) cornerRadius else CornerRadius.Zero,
+                                topRight = if (index == textBounds.lastIndex) cornerRadius else CornerRadius.Zero,
+                                bottomRight = if (index == textBounds.lastIndex) cornerRadius else CornerRadius.Zero
+                            )
+                        )
+                    }
+                    drawPath(
+                        path = path,
+                        color = highlightColor,
+                        style = Fill
+                    )
+                    if (highlightBorderColor != null) {
+                        drawPath(
+                            path = path,
+                            color = highlightBorderColor,
+                            style = Stroke(width = highlightBorderWidth.toPx())
```

Comment:
`Stroke(width = highlightBorderWidth.toPx())` can be placed above under a `remember`